### ISSUE TITLE
Remove `isValidState` utility

### ISSUE THREAD

**Issue by marip8** [tesseract-robotics/tesseract_planning#325]
https://github.com/tesseract-robotics/tesseract_planning/blob/9128a70ca0d7606394c4d9949813f5eadea34f98/tesseract_motion_planners/core/include/tesseract_motion_planners/planner_utils.h#L72-L93

I was just perusing through some of the planner utility functions and I found the one above that seems very tied to a specific robot configuration. I think we should remove this function as it could be confusing to users

**Comment by Levi-Armstrong** [comment]
I don't think it should be removed but would be good to update the documentation so it is clear when to use this utility method.
